### PR TITLE
fix: improve api/cron/webhookTriggers endpoint

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

The endpoint was temporarily failing because it timed out. This PR introduces some performance improvements

### CODE REVIEW

File: packages/features/webhooks/lib/cron.ts [hunk 1]
```diff
@@ -31,20 +31,35 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
...
-    let webhook;
-    try {
-      webhook = await prisma.webhook.findUniqueOrThrow({
-        where: { id: subscriberId, appId: appId !== "null" ? appId : null },
-      });
-    } catch {
-      logger.error(`Error finding webhook for subscriberId: ${subscriberId}, appId: ${appId}`);
+    if (!webhook) {
+      try {
+        webhook = await prisma.webhook.findUniqueOrThrow({
```
Comment (CarinaWolli) [inline packages/features/webhooks/lib/cron.ts:57]: We now save the webhook to `webhookScheduledTriggers` so we don't have to do that anymore for each job

File: packages/features/webhooks/lib/cron.ts [hunk 2]
```diff
@@ -56,33 +71,15 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
...
+      fetch(job.subscriberUrl, {
         method: "POST",
         body: job.payload,
         headers,
-      });
-    } catch (error) {
-      console.log(`Error running webhook trigger (retry count: ${job.retryCount}): ${error}`);
-
-      // if job fails, retry again for 5 times.
-      if (job.retryCount < 5) {
```
Comment (CarinaWolli) [inline packages/features/webhooks/lib/cron.ts:69]: retry was not implemented correctly and it returned `ok: false` on the first error. This also caused that all the other jobs were not triggered

I don't think we need to retry if the request fails, now we are just login the error and continue with the other jobs

File: packages/features/webhooks/lib/cron.ts [hunk 3]
```diff
@@ -122,6 +119,8 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
     });
   }
 
+  Promise.allSettled(fetchPromises);
```
Comment (CarinaWolli) [inline packages/features/webhooks/lib/cron.ts:122]: process fetch operations asynchronously

**Review by joeauyeung — APPROVED** [review]
PR LGTM.

I wonder if 5 retries was too generous and that was causing the timeouts.

**Review by zomars — APPROVED** [review]
LGTM!

**Pull request merged**